src/FileFormats/Reader.cs

```diff
@@ -22,6 +23,30 @@ public Reader(IAddressSpace dataSource, LayoutManager layoutManager)
         public LayoutManager LayoutManager { get; private set; }
         public IAddressSpace DataSource { get; private set; }
 
+        public string ReadCountedString(ulong position, Encoding encoding)
```

Comment:
I'm hoping to leave Reader very simple, otherwise I worry it will fill with too many ReadXYZ() helpers that are scenario specific (for example what if another scenario has a counted string with a 2 byte length instead of 4?) As an alternative I would suggest either:
a) Create a static class with extension methods, callers wouldn't need to change at all
b) Create a type CountedStringLayout that is similar to NullTerminatedStringLayout. 
     i) Only one layout can be used when you call reader.Read<string>( ... ), but you can choose which one it is when you set up your LayoutManager
     ii) OR you could create a type CountedString that has an implicit cast to string and do string s = reader.Read<CountedString>( ... )

Right now there isn't much advantage to doing the layout unless you particularly like the Read<T>(...) syntax, however in the future you could imagine something like a TStruct or array that composes over types that aren't fixed size. I'd probably go with the static extension method for now, but either one works.